src/utils/rules.ts

```diff
@@ -38,21 +39,25 @@ export const styleColor = ({ prop, cssProperty }: { prop: string; cssProperty?:
       val = get(props.theme, val);
     } else {
       // is taken from a prop
-      if (typeof val === 'string') {
-        val = val.split('.');
-      }
+      if (val.startsWith('@')) {
+        if (typeof val === 'string') {
```

Comment:
This condition here doesn't make much sense to me, since at line 42 `val.startsWith('@')` we already assume that val is indeed a string. If we are not certain whether val is a string, let's move that condition above to avoid TypeErrors.